mrblib/shelf/dispatcher.rb

```diff
@@ -43,8 +43,9 @@ def initialize(map = {})
     def call(env)
       path   = env[PATH_INFO]
       method = R3.method_code(env[REQUEST_METHOD])
+      raise ArgumentError, "Invalid HTTP method: #{env[REQUEST_METHOD].inspect}." unless method
```

Comment:
Without this, if we get invalid HTTP request method, Shelf raises confusing error like `TypeError: can't convert nil to Integer`.

Reply:
that should also result in an `HTTP 405` response instead of an server exception